File: AIペアプログラミング実践編_ソースコード/practical-java/session-5/Ans/src/main/java/com/example/inventory/repository/OrderRepository.java
package com.example.inventory.repository;

import com.example.inventory.entity.Order;
import com.example.inventory.entity.OrderStatus;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * 発注リポジトリ
 */
@Repository
public interface OrderRepository extends JpaRepository<Order, Long> {
    List<Order> findByStatus(OrderStatus status);
}

File: AIペアプログラミング実践編_ソースコード/practical-csharp/session-5/Ans/Inventory.Api/Repositories/OrderRepository.cs
namespace Inventory.Api.Repositories;

using Inventory.Api.Data;
using Inventory.Api.Entities;

public class OrderRepository : IOrderRepository
{
    private readonly InventoryDbContext _context;

    public OrderRepository(InventoryDbContext context)
    {
        _context = context;
    }

    public Order? FindById(int id)
    {
        return _context.Orders.Find(id);
    }

    public List<Order> FindAll()
    {
        return _context.Orders.ToList();
    }

    public List<Order> FindByStatus(OrderStatus status)
    {
        return _context.Orders.Where(o => o.Status == status).ToList();
    }

    public Order Save(Order order)
    {
        if (order.Id == 0)
        {
            _context.Orders.Add(order);
        }
        else
        {
            _context.Orders.Update(order);
        }
        _context.SaveChanges();
        return order;
    }
}
